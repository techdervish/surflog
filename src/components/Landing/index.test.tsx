import {
  expectWrappersTreeToMatchSnapshot,
  createShallowWrapper,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('Landing component', () => {
  const mockOnConfirmed = jest.fn();
  const mockClickInfo = jest.fn();

  const wrapper = createWrapper({
    onConfirmed: mockOnConfirmed,
    onClickInfo: mockClickInfo,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('should call onClickInfo prop on click', () => {
    wrapper
      .find('a')
      .first()
      .simulate('click');
    expect(mockClickInfo).toBeCalled();
  });

  it('should call onConfirmed prop on click', () => {
    wrapper
      .find('a')
      .last()
      .simulate('click');
    expect(mockOnConfirmed).toBeCalled();
  });
});
