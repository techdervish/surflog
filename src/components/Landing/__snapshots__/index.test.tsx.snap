// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders the expected snapshot 1`] = `
<div>
  <div
    className="f4 fw2 lh-title pt3"
  >
    <div
      className="f2 tc"
    >
      <span
        className="fw3"
      >
        Browser history with content, works for
      </span>
       
      <RecordedPagesSlide />
    </div>
    <div
      className="pt3 f5"
    >
      <label
        className="fw5"
      >
        How does it work?
      </label>
      <div>
        A new record is created for new visits or for the interactions on supported websites. A record consists of metadata (time, link, ..) and the required content extracted from website.
         
        <div
          className="pt2"
        >
          All records are kept 
          <b>
            in your browser's storage
          </b>
          , so that you can search your records pretty effectively
        </div>
      </div>
    </div>
    <div
      className="pt3 f5"
    >
      <label
        className="fw5"
      >
        Is it safe?
      </label>
      <div>
        All data is stored in your device, nothing is sent or shared. All the code can be inspected and you can learn more on
         
        <a
          className="black fw4"
          href="#"
          onClick={[MockFunction]}
        >
          Github
        </a>
        .
      </div>
    </div>
  </div>
  <a
    className="mt3 f5 no-underline black bg-animate hover-bg-black hover-white inline-flex items-center pa3 ba border-box mr4 justify-center w-100"
    href="#0"
    onClick={[MockFunction]}
  >
    SOUNDS GOOD
  </a>
</div>
`;
